src/libcore/iter.rs

```diff
@@ -369,6 +369,25 @@ pub trait Iterator {
     /// `None` here means that either there is no known upper bound, or the
     /// upper bound is larger than `usize`.
     ///
+    /// # Safety
+    ///
+    /// Implementations of this trait are not guaranteed to yield
+    /// declared number of elements. Iterator may yield less than lower
```

Comment:
This first sentence sounds a bit stunted to me. I think "yield the declared" would help, as well as "The iterator may" in the second one.